client/src/backoffice/Main.tsx

```diff
@@ -16,53 +16,115 @@ import { AddressManagementPage } from './AddressManagementPage';
 import { TransactionsPage } from './TransactionsPage';
 import { InboxPage } from './InboxPage';
 import { InboxListPage } from './InboxListPage';
+import { QrPage } from './QrPage';
+import Calendar from '../images/calendar.svg';
+import Qr from '../images/qr-code.svg';
 
 export const MintersPage = () => <div> This is a MintersPage </div>;
 
+type RouteProps = {
+  path: string;
+  roles: string[];
+  title?: string;
+};
+
+type LabelProps = {
+  roles: string[];
+  title: string;
+};
+
+const Label: React.FC<{ label: LabelProps }> = ({ label }) => {
+  const { title } = label;
+  return <h2>{title}</h2>;
+};
+
+const RoleLink: React.FC<{ route: RouteProps; handleClick: () => void }> = ({
+  route,
+  handleClick,
+}) => {
+  const { path, title } = route;
+  if (typeof route === 'object' && title) {
+    return (
+      <Link className={'bm-item'} to={path} onClick={handleClick}>
+        {title}
+      </Link>
+    );
+  }
+
+  return null;
+};
+
+const WithRole: React.FC<{ roles: string[] }> = ({ roles, children }) => {
+  // TODO: Get user role (Backend WIP)
+  const userRole = 'super';
+
+  if (!roles.includes(userRole)) return null;
+
+  return <>{children}</>;
+};
```

Comment:
Hi @niconiahi I want to change this component to a HOC so we don't have to wrap all components that needs to check for role like this:
```jsx
<WithRole roles={ROUTES.issueForEvent.roles}>
  <RoleLink route={ROUTES.issueForEvent} handleClick={closeMenu} />
</WithRole>
```

Implementation of a WithRoleHOC (word HOC on the definition isn't necessary):

```jsx
type Roles = {
  roles: string[];
};

function WithRoleHOC<Props>(WrappedComponent: React.ComponentType<Props>) {
  const hocComponent = (props: Props & Roles) => {
    const userRole = 'super';
    if (!props.roles.includes(userRole)) return null;

    return <WrappedComponent {...props} />;
  };

  return hocComponent;
}
```

Usage of WithRoleHOC:

```typescript
const LabelWithRole = WithRoleHOC(Label);
const LinkWithRole = WithRoleHOC(RoleLink);

// In render:
<LabelWithRole roles={LABELS.issueBadges.roles} label={LABELS.issueBadges}/> 

<LinkWithRole
  handleClick={closeMenu}
  route={ROUTES.issueForEvent}
  roles={ROUTES.issueForEvent.roles}
/>
```

Reply:
usage done, missing props typing

Reply:
Great @niconiahi don't forget about https://github.com/Xivis/poap/pull/42#discussion_r350188822